src/hooks/useAppMutation.ts

```diff
@@ -0,0 +1,48 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
```

Comment:
Nice work on the `useAppMutation` wrapper! For consistency, creating a similar `useAppQuery` abstraction to handle `neverthrow` results and error toasting should be considered. While you  only have 2 query usages currently, this would keep data fetching patterns uniform.